### ISSUE TITLE
Specify AWS profile instead of credentials

### ISSUE THREAD

**Issue by katspaugh** [serverless/serverless#392]
In JAWS – in `admin.env` – you just said which AWS profile to use – and it was very convnient.
In Serverless `admin.env` needs to contain AWS credentials. 😔

With Serverless I have to do this on CircleCI:

```
deployment:
  dev:
    branch: master
    commands:
      - cat ~/.aws/credentials | grep aws_access_key_id | sed 's/aws_access_key_id = /SERVERLESS_ADMIN_AWS_ACCESS_KEY_ID=/g' > admin.env
      - cat ~/.aws/credentials | grep aws_secret_access_key | sed 's/aws_secret_access_key = /SERVERLESS_ADMIN_AWS_SECRET_ACCESS_KEY=/g' >> admin.env
      - serverless resources deploy
```

**Comment by kennu** [comment]
Same problem. Previously our team could just keep admin.env in Git, as everybody already has many AWS profiles configured on all their computers. JAWS would then automatically use the right profile without any setup.

**Comment by doapp-ryanp** [comment]
+1 to being able to use profile vs credentials.  

Many devs will have lots of projects, so not only are the points above valid, but also the need to update creds in one spot (`~/.aws/credentials`).

Also the file perms on `~/.aws` will typically be more restrictive by default than the perms of `admin.env`.  People who do not know the fallout of not protecting their `admin.env` could get hurt pretty bad.  In addition of `.gitignore` got accidentally modified and `admin.env` got checked in, could be catastrophic. Where accidentally checking in `admin.env` with just a profile name would not be.

**Comment by visualasparagus** [comment]
+1

**Comment by JamieCressey** [comment]
+1

**Comment by apaatsio** [comment]
+1

**Comment by joostfarla** [comment]
> Previously our team could just keep admin.env in Git, as everybody already has many AWS profiles configured on all their computers. JAWS would then automatically use the right profile without any setup.

+1

**Comment by obie** [comment]
+1

**Comment by martinlindenberg** [comment]
I have a solution for this problem. can anyone put this into the 0.0.15 ?

it is only a small change in the file Serverless.js. 

it checks the process.env.SERVERLESS_ADMIN_AWS_PROFILE and automatically fills up the attributes this._awsAdminKeyId and this._awsAdminSecretKey
[Serverless.js.txt](https://github.com/serverless/serverless/files/79508/Serverless.js.txt)

Changed Lines:
line 8: added awsMisc
line 52: use a function to set the attributes
line 68: function added

**Comment by katspaugh** [comment]
Resolved in #462.
